File: src/main/java/no/hvl/tk/visual/debugger/DebugProcessListener.java
package no.hvl.tk.visual.debugger;

import com.intellij.xdebugger.XDebugProcess;
import com.intellij.xdebugger.XDebugSession;
import com.intellij.xdebugger.XDebuggerManagerListener;
import no.hvl.tk.visual.debugger.debugging.stackframe.StackFrameSessionListener;
import org.jetbrains.annotations.NotNull;

public class DebugProcessListener implements XDebuggerManagerListener {

    @Override
    public void processStarted(@NotNull final XDebugProcess debugProcess) {
        final XDebugSession debugSession = debugProcess.getSession();
        debugSession.addSessionListener(new StackFrameSessionListener(debugProcess));
    }
}

File: src/main/java/no/hvl/tk/visual/debugger/DebugProcessListener.kt
package no.hvl.tk.visual.debugger

import com.intellij.xdebugger.XDebugProcess
import com.intellij.xdebugger.XDebuggerManagerListener
import no.hvl.tk.visual.debugger.debugging.stackframe.StackFrameSessionListener

class DebugProcessListener : XDebuggerManagerListener {
  override fun processStarted(debugProcess: XDebugProcess) {
    val debugSession = debugProcess.session
    debugSession.addSessionListener(StackFrameSessionListener(debugProcess))
  }
}
